src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -2,143 +2,180 @@
             }
-
-            if (_relativeRootSolutionFolders != null)
+            // Generate intermediate solution folders
+            if (solutionFolder is null && !_inRoot)
             {
-                // The user has specified an explicit root
-                return _relativeRootSolutionFolders;
+                var relativePathDirectory = Path.GetDirectoryName(pathRelativeToSolutionFile);
+                if (!string.IsNullOrEmpty(relativePathDirectory))
+                {
+                    SolutionFolderModel relativeSolutionFolder = solution.AddFolder(GetSolutionFolderPathWithForwardSlashes(relativePathDirectory));
+                    project.MoveToFolder(relativeSolutionFolder);
+                    // Avoid duplicate folder/project names
+                    if (project.Parent is not null && project.Parent.ActualDisplayName == project.ActualDisplayName)
+                    {
+                        solution.RemoveFolder(project.Parent);
+                    }
+                }                
             }
+            // Add settings based on existing project instance
+            ProjectInstance projectInstance = new ProjectInstance(projectRootElement);
+            string projectInstanceId = projectInstance.GetProjectId();
+            if (!string.IsNullOrEmpty(projectInstanceId))
+            {
+                project.Id = new Guid(projectInstanceId);
+            }
+            var projectInstanceBuildTypes = projectInstance.GetConfigurations();
+            var projectInstancePlatforms = projectInstance.GetPlatforms();
 
-            // We determine the root for each individual project
-            var relativeProjectPath = Path.GetRelativePath(
-                PathUtility.EnsureTrailingSlash(slnFile.BaseDirectory),
-                fullProjectPath);
-
-            return GetSolutionFoldersFromProjectPath(relativeProjectPath);
-        }
-
-        private static bool IsPathInTreeRootedAtSolutionDirectory(string path)
-        {
-            return !path.StartsWith("..");
-        }
-
-        private static string TrimProject(string path)
-        {
-            return Path.GetDirectoryName(path);
-        }
+            foreach (var solutionPlatform in solution.Platforms)
+            {
+                var projectPlatform = projectInstancePlatforms.FirstOrDefault(
+                    platform => platform.Replace(" ", string.Empty) ==  solutionPlatform.Replace(" ", string.Empty), projectInstancePlatforms.FirstOrDefault("Any CPU"));
+                project.AddProjectConfigurationRule(new ConfigurationRule(BuildDimension.Platform, "*", solutionPlatform, projectPlatform));
+            }
 
-        private static string TrimProjectDirectory(string path)
-        {
-            return Path.GetDirectoryName(path);
+            foreach (var solutionBuildType in solution.BuildTypes)
+            {
+                var projectBuildType = projectInstanceBuildTypes.FirstOrDefault(
```

Comment:
Is debug what we want for the default behavior? Is that documented anywhere?

Reply:
You are right. This is repetitive as we are adding the default build types anyways and "Debug" is the first one

Reply:
Good point, thank you.